pkgs/tools/security/cdxgen/default.nix

```diff
@@ -1,26 +1,66 @@
-{ lib
-, fetchFromGitHub
-, buildNpmPackage
+{ fetchFromGitHub
+, lib
+, makeWrapper
+, nodejs
+, nodePackages
+, pnpm
+, python3
+, stdenv
+, xcbuild
 }:
 
-buildNpmPackage rec {
+stdenv.mkDerivation rec {
   pname = "cdxgen";
-  version = "10.5.2";
+  version = "10.8.1";
 
   src = fetchFromGitHub {
-    owner = "AppThreat";
+    owner = "CycloneDX";
     repo = pname;
     rev = "v${version}";
-    sha256 = "sha256-CmX19UdmXTbmO+6nFzsFbZspmIWYFtcUVaA0j8iU7GI=";
+    hash = "sha256-PFvSHuIaHaGfKI5s7DOW1adSKpnURaQtk5lAO9lr1OM=";
   };
 
-  npmDepsHash = "sha256-Vd+zRExQFmmv9f8uWQFE/nWRs6y86nLFu5HrM6iCf7U=";
+  nativeBuildInputs = [
+    makeWrapper
+    nodejs
+    nodePackages.node-gyp # required for sqlite3 bindings
+    pnpm.configHook
+    python3 # required for sqlite3 bindings
+  ] ++ lib.optional stdenv.isDarwin [ xcbuild ];
+
+  pnpmDeps = pnpm.fetchDeps {
+    inherit pname version src;
+    hash = "sha256-IO7hn5xHdlQ+uyH8RWc7ZnnthXydCnMSde22YYMWOoc=";
+  };
+
+  buildPhase = ''
+    runHook preBuild
+
+    pushd node_modules/sqlite3
+    node-gyp rebuild
+    popd
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin $out/lib
+    cp -r * $out/lib
+    makeWrapper ${nodejs}/bin/node "$out/bin/cdxgen" --add-flags "$out/lib/bin/cdxgen.js"
+    makeWrapper ${nodejs}/bin/node "$out/bin/evinse" --add-flags "$out/lib/bin/evinse.js"
+    makeWrapper ${nodejs}/bin/node "$out/bin/repl" --add-flags "$out/lib/bin/repl.js"
+    makeWrapper ${nodejs}/bin/node "$out/bin/verify" --add-flags "$out/lib/bin/verify.js"
```

Comment:
I am not sure what the other binaries do. But I would name them like `cdxgen-evinse` etc.